### PR TITLE
feature/select multiple entity actions

### PR DESCRIPTION
**Pull request by talohana**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

1. `store.remove()` emits `ids` as null
2. `selectEntityAction` accepts single or none `EntityActions`

Issue Number: #501

## What is the new behavior?
1. `store.remove()` emits `ids` all ids
2. `selectEntityAction` accepts single, multiple or none `EntityActions`

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

File: libs/akita/src/lib/queryEntity.ts [hunk 1]
```diff
@@ -291,22 +291,24 @@ export class QueryEntity<S extends EntityState, EntityType = getEntityType<S>, I
...
+  selectEntityAction(...actions: EntityActions[]): Observable<{ [Name in EntityActionsNames]?: IDType[] }>;
   selectEntityAction(): Observable<EntityAction<IDType>>;
-  selectEntityAction(action?: EntityActions): Observable<IDType[] | EntityAction<IDType>> {
-    if (isUndefined(action)) {
+  selectEntityAction(...actions: EntityActions[]): Observable<{ [Name in EntityActionsNames]?: IDType[] } | EntityAction<IDType>> {
+    if (isEmpty(actions)) {
       return this.store.selectEntityAction$;
     }
+
     return this.store.selectEntityAction$.pipe(
```
Comment (NetanelBasal) [inline libs/akita/src/lib/queryEntity.ts:309]: This is a breaking change @xbaun

File: libs/akita/src/__tests__/entityStore.spec.ts [hunk 2]
```diff
@@ -349,8 +349,8 @@ describe('EntitiesStore', () => {
       store.add(todo);
       store.add(todo2);
       store.remove();
-      expect(store.entities[1]).toBeUndefined();
-      expect(store.entities[2]).toBeUndefined();
+      expect(store._value().entities[1]).toBeUndefined();
```
Comment (NetanelBasal) [inline libs/akita/src/__tests__/entityStore.spec.ts:352]: Why did you change everything to _value()?

Comment (talohana) [reply libs/akita/src/__tests__/entityStore.spec.ts:352]: Had compilation error becahse `store.entities` is private, I saw other suites used `store._value().entities` instead

File: libs/akita/src/lib/queryEntity.ts (hunk 1 above)
Comment (xbaun) [reply libs/akita/src/lib/queryEntity.ts:309]: The problem (if it is) is how to distinguish between results of different entity actions. On the other hand, we could of course simply return the ids as we do now and not distinguish between them.

Comment (NetanelBasal) [reply libs/akita/src/lib/queryEntity.ts:309]: Another option is to overload it an accept an array instead.

Comment (xbaun) [reply libs/akita/src/lib/queryEntity.ts:309]: I think we were talking about different things: I meant that the value of the subscription is a breaking change. To overload the function for backwards compatibility is the other.

Comment (NetanelBasal) [reply libs/akita/src/lib/queryEntity.ts:309]: Yes, that's what I meant when saying a braking change.

Comment (talohana) [reply libs/akita/src/lib/queryEntity.ts:309]: Just to make it clear, the function will return `ids` array for single type, and `{ [EntityActions.Type]: ids }` for array input?

Comment (xbaun) [reply libs/akita/src/lib/queryEntity.ts:309]: @talohana now I get it, sounds good 👍

Comment (NetanelBasal) [reply libs/akita/src/lib/queryEntity.ts:309]: Yes, that's what I meant.

Comment (talohana) [reply libs/akita/src/lib/queryEntity.ts:313]: I've updated accordingly, just not 100% the function return type is the best I can get out of it.

File: libs/akita/src/lib/queryEntity.ts [hunk 3]
```diff
@@ -291,22 +291,28 @@ export class QueryEntity<S extends EntityState, EntityType = getEntityType<S>, I
...
    *  this.query.selectEntityAction(EntityActions.Add);
    *  this.query.selectEntityAction(EntityActions.Update);
    *  this.query.selectEntityAction(EntityActions.Remove);
    *
+   *  this.query.selectEntityAction([EntityActions.Add, EntityActions.Update, EntityActions.Remove])
+   *
    *  this.query.selectEntityAction();
    */
   selectEntityAction(action: EntityActions): Observable<IDType[]>;
+  selectEntityAction(actions: EntityActions[]): Observable<IDType[]>;
```
Comment (xbaun) [inline libs/akita/src/lib/queryEntity.ts:303]: Should be `  selectEntityAction(actions: EntityActions[]): Observable<EntityAction<IDType>>;`

File: libs/akita/src/lib/queryEntity.ts [hunk 4]
```diff
@@ -291,22 +291,28 @@ export class QueryEntity<S extends EntityState, EntityType = getEntityType<S>, I
...
   selectEntityAction(): Observable<EntityAction<IDType>>;
-  selectEntityAction(action?: EntityActions): Observable<IDType[] | EntityAction<IDType>> {
-    if (isUndefined(action)) {
+  selectEntityAction(actionOrActions?: EntityActions | EntityActions[]): Observable<IDType[] | { [Name in EntityActionsNames]?: IDType[] } | EntityAction<IDType>> {
+    if (isNil(actionOrActions)) {
       return this.store.selectEntityAction$;
     }
+
+    const project = isArray(actionOrActions) ? ({ type, ids }: EntityAction<IDType>) => ({ [type]: ids }) : ({ ids }: EntityAction<IDType>) => ids;
```
Comment (xbaun) [inline libs/akita/src/lib/queryEntity.ts:310]: @NetanelBasal Instead of rewriting the `EntityAction` we should just pass it through?
```
    const project = isArray(actionOrActions) ? ((entityAction: EntityAction<IDType>) => entityAction) : ({ ids }: EntityAction<IDType>) => ids;
```

Comment (NetanelBasal) [reply libs/akita/src/lib/queryEntity.ts:310]: Yes

**Comment by NetanelBasal** [comment]
Please update the docs

**Comment by talohana** [comment]
> Please update the docs

Working on it right now, is there an easy way to run the docs site? couldn't find a script for it.

**Comment by NetanelBasal** [comment]
It is in the docs folder
